### PR TITLE
sr3 status fixes #1350 and #1352. 

### LINKED ISSUE #1350: Another run state: disconnected
I think an additional run state that indicates connection failures to file transfer services (i.e. not brokers) might be useful, particularly for senders or polls.

If we can't connect to the `sendTo` or `pollUrl`, then show this state. We would know right away that there's a source or destination issue.

Log monitoring would also indicate source/destination problems, but this might be faster to implement.

I'm pretty sure `retry` is the state that would currently be shown in these situations, but an explicit connection error state might be easier to understand.

### LINKED ISSUE #1352: post_retry counts are not included in sr3 status *retry*
during flow tests:
```
sarra@edcm-dirt18-1:~/.cache/sr3$ wc -l */*/*retry*
    412 poll/sftp_f62/diskqueue_post_retry_001
      0 poll/sftp_f62/diskqueue_post_retry_001.new
    124 poll/sftp_f63/diskqueue_post_retry_001
      0 poll/sftp_f63/diskqueue_post_retry_001.new
    510 post/t_dd1_f00/diskqueue_post_retry_000
      0 post/t_dd1_f00/diskqueue_post_retry_000.new
    551 post/t_dd2_f00/diskqueue_post_retry_000
      0 post/t_dd2_f00/diskqueue_post_retry_000.new
     17 post/test2_f61/diskqueue_post_retry_000
      0 post/test2_f61/diskqueue_post_retry_000.new
     10 sarra/download_f20/diskqueue_post_retry_001
      0 sarra/download_f20/diskqueue_post_retry_001.new
     74 sender/tsource2send_f50/diskqueue_post_retry_001
      0 sender/tsource2send_f50/diskqueue_post_retry_001.new
     74 shovel/rabbitmqtt_f22/diskqueue_post_retry_001
      0 shovel/rabbitmqtt_f22/diskqueue_post_retry_001.new
     48 watch/f40/diskqueue_post_retry_001
      0 watch/f40/diskqueue_post_retry_001.new
   1820 total
sarra@edcm-dirt18-1:~/.cache/sr3$ sr3 status
status:
Component/Config         Processes                                         Rates
                         State   Run Retry  Que     Lag   Last    %rej  messages      Data
                         -----   --- -----  ---     ---   ----    ----  --------      ----
cpost/pelle_dd1_f04      stop    0/0    0    0    0.00s      0    0.0%     0m/s       0B/s
cpost/pelle_dd2_f05      stop    0/0    0    0    0.00s      0    0.0%     0m/s       0B/s
cpost/veille_f34         idle    1/1    0    0    0.00s    n/a    0.0%     0m/s       0B/s
cpump/xvan_f14           idle    1/1    0    0    0.00s    n/a    0.0%     0m/s       0B/s
cpump/xvan_f15           idle    1/1    0    0    0.00s    n/a    0.0%     0m/s       0B/s
poll/sftp_f62            wVip    1/1    0    0    0.00s  3m55s    0.0%     0m/s       0B/s
poll/sftp_f63            run     1/1    0    0    0.00s  3m19s    0.0%     0m/s       0B/s
post/shim_f63            stop    0/0    -    -       -      -       -       -           -
post/t_dd1_f00           stop    0/0    0    0    0.00s      0    0.0%     0m/s       0B/s
post/t_dd2_f00           stop    0/0    0    0    0.00s  2m20s    0.0%     0m/s       0B/s
post/test2_f61           stop    0/0    0    0    0.00s  4m44s    0.0%     0m/s       0B/s
report/tsarra_f20        idle    1/1    0    0    0.00s    n/a    0.0%     0m/s       0B/s
sarra/download_f20       idle    1/1    0    0    0.00s      0    0.0%     0m/s       0B/s
sender/tsource2send_f50  idle    1/1    0    0    0.00s  5m25s    0.0%     0m/s       0B/s
shovel/rabbitmqtt_f22    hung    1/1    0    0    0.00s  5m25s    0.0%     0m/s       0B/s
subscribe/amqp_f30       idle    1/1    0    0    0.00s    n/a    0.0%     0m/s       0B/s
subscribe/cdnld_f21      idle    1/1    0    0    0.00s    n/a    0.0%     0m/s       0B/s
subscribe/cfile_f44      idle    1/1    0    0    0.00s    n/a    0.0%     0m/s       0B/s
subscribe/cp_f61         idle    1/1    0    0    0.00s    n/a    0.0%     0m/s       0B/s
subscribe/ftp_f70        idle    1/1    0    0    0.00s    n/a    0.0%     0m/s       0B/s
subscribe/q_f71          idle    5/5    0    0    0.00s    n/a    0.0%     0m/s       0B/s
subscribe/rabbitmqtt_f31 idle    1/1    0    0    0.00s    n/a    0.0%     0m/s       0B/s
subscribe/u_sftp_f60     idle    1/1    0    0    0.00s    n/a    0.0%     0m/s       0B/s
watch/f40                run     1/1    0    0    0.00s  5m26s    0.0%     0m/s       0B/s
      Total Running Configs:  18 ( Processes: 22 missing: 0 stray: 0 )
                     Memory: uss:748.8MiB rss:996.7MiB vms:4.0GiB
                   CPU Time: User:162.76s System:11.90s
           Pub/Sub Received: 0m/s (189B/s), Sent:  0.39m/s (189B/s) Queued: 0 Retry: 0, Mean lag: 0.00s
              Data Received: 0f/s (0B/s), Sent: 0f/s (0B/s)
sarra@edcm-dirt18-1:~/.cache/sr3$

```

it looks like the metrics of sr.py CLI isn't counting the stuff in the post_retry queues
under *retry* column... so it looks clean in sr3 status, but there's actually 100's of retries pending.

### PR DESCRIPTION
**Pull request by petersilva**
close #1352 
close #1350 

For #1350, there was a tally missing from empry_metrics so that the post_retries were not being counted.

For #1350, added two states, *disconnected* is for when communications with the broker is broken. *down* is when data transfer connection is broken.

seems to work.

### CODE REVIEW

File: sarracenia/sr.py [hunk 1]
```diff
@@ -55,13 +55,14 @@
...
-        "lagMean": 0, "latestTransfer": 0, "rejectPercent":0, "transferRxByteRate":0, "transferTxByteRate": 0,
+empty_metrics={ "byteRate":0, "cpuTime":0, "connected": True, "rejectCount":0, "last_housekeeping":0, "messagesQueued": 0, 
+               "lagMean": 0, "latestTransfer": 0, "rejectPercent":0, "transferConnected": True, "transferRxByteRate":0, 
+               "transferTxByteRate": 0,
         "rxByteCount":0, "rxGoodCount":0, "rxBadCount":0, "txByteCount":0, "txGoodCount":0, "txBadCount":0, 
         "lagMax":0, "lagTotal":0, "lagMessageCount":0, "disconnectTime":0, "transferConnectTime":0, 
         "transferRxLast": 0, "transferTxLast": 0, "rxLast":0, "txLast":0, 
         "transferRxBytes":0, "transferRxFiles":0, "transferTxBytes": 0, "transferTxFiles": 0, 
-        "msgs_in_post_retry": 0, "msgs_in_download_retry":0, "brokerQueuedMessageCount": 0, 
+               "msgs_in_post_retry": 0, "msgs_in_download_retry":0, "msgs_in_post_retry":0, "brokerQueuedMessageCount": 0,
```
Comment (MagikEh) [inline sarracenia/sr.py]: What's with all the whitespace at the start here?

Comment (MagikEh) [inline sarracenia/sr.py]: There are a few lines of misalignment..

**Review by MagikEh — APPROVED** [review]
✅ Looks great! RS also signed off on the logics ;)

**Review by reidsunderland — APPROVED** [review]
image

**Comment by MagikEh** [comment]
Very alphabet

(google owns me)

**Pull request merged**